### PR TITLE
Result.py: Adds condition on confidence in Result

### PR DESCRIPTION
**Pull request by arush0311**
Fixes https://github.com/coala-analyzer/coala/issues/2605

### CODE REVIEW

**Comment by arush0311** [comment]
Oops! No tests added, working on it.

File: tests/results/ResultTest.py [hunk 1]
```diff
@@ -24,6 +24,10 @@ def test_invalid_severity(self):
         with self.assertRaises(ValueError):
             Result("o", "m", severity=-5)
 
+    def test_invalid_confidence(self):
+        with self.assertRaises(ValueError):
+            Result("o", "m", confidence=-1)
```
Comment (Makman2) [inline tests/results/ResultTest.py]: I would also add a test if the confidence is above 100 :)

**Comment by Makman2** [comment]
ack 1ba90b6

**Comment by Makman2** [comment]
only rebase needed, then we can merge :D

**Comment by SanketDG** [comment]
could you rebase one more time?

**Comment by Makman2** [comment]
oh wait found a slight issue: could you change the commit message a bit?
>  Result~~.py~~: Add~~s~~ condition on confidence~~ in Result~~

1. we don't supply the .py ending, saves space for the commit message :)
2. we use present tense in commit messages
3. "in Result" is not needed, that's what the tag is for :D

sorry I missed that :3

**Comment by Udayan12167** [comment]
ack 14708f4

File: coalib/results/Result.py [hunk 2]
```diff
@@ -70,7 +71,10 @@ def __init__(self,
         # Sorting is important for tuple comparison
         self.affected_code = tuple(sorted(affected_code))
         self.severity = severity
-        self.confidence = confidence
+        if confidence < 0 or confidence > 100:
+            raise ValueError('Value of confidence should be between 0 and 100.')
+        else:
+            self.confidence = confidence
```
Comment (sils1297) [inline coalib/results/Result.py]: hey, normally you wouldn't do an else here, @Udayan12167 can you file a newcomer issue to remove the else?

Comment (Udayan12167) [inline coalib/results/Result.py]: Okay sure.

**Comment by Udayan12167** [comment]
@rultor merge

**Pull request merged**